lib/fpm/package/python.rb

```diff
@@ -195,8 +210,185 @@ def download_if_necessary(package, version=nil)
+      opt = "off"
+      flags += [ "--progress-bar", opt]
+      flags += [ "--no-deps" ]
+      flags += [ "--use-pep517" ]
+      flags += [ "--check-build-dependencies" ]
+
+      # @todo FIXME!!! --config-settings for PEP 517 build backend (KEY=VALUE, can be multiple)
+      # I have no clue where to get all that 'options' and any description of possible 'backends'
+      if !attributes[:python_build_backend_arguments].nil? and !attributes[:python_build_backend_arguments].empty?
+        # Add optional arguments for pep517 build backend
+        attributes[:python_build_backend_arguments].each do |a|
+          flags += [ "--config-settings", a ]
+        end
+      end
+
+      safesystem(*attributes[:python_pip], "wheel", ".", *flags)
+    end
+
+    files = ::Dir.glob(File.join(project_dir, wheel_dir, "*.whl"))
+    if files.length != 1
+      raise "Unexpected directory layout after `pip wheel ...`. This might be an fpm bug? The directory is #{build_path}"
+    end
+
+    return files[0]
+  end # def build_py_wheel
+
+  # Load the package information like name, version, dependencies from wheel file.
+  def load_package_info_wheel(setup_data, wheel_path)
+    if !attributes[:python_package_prefix].nil?
+      attributes[:python_package_name_prefix] = attributes[:python_package_prefix]
+    end
+
+    begin
+      safesystem("#{attributes[:python_bin]} -c 'import json'")
+    rescue FPM::Util::ProcessFailed => e
+      logger.error("Your python environment is missing json support. I cannot continue without this.", :python => attributes[:python_bin], :error => e)
+      raise FPM::Util::ProcessFailed, "Python (#{attributes[:python_bin]}) is missing json modules."
+    end
+
+    begin
+      safesystem("#{attributes[:python_bin]} -c 'from pkginfo import Wheel'")
+    rescue FPM::Util::ProcessFailed => e
+      logger.error("Your python environment is missing a working pkginfo.Wheel module. I tried to find the 'importlib.Wheel' module but failed.", :python => attributes[:python_bin], :error => e)
+      raise FPM::Util::ProcessFailed, "Python (#{attributes[:python_bin]}) is missing pkginfo.Wheel module."
+    end
+
+    # Add ./pyfpm_wheel/ to the python library path
+    pylib = File.expand_path(File.dirname(__FILE__))
+
+    # chdir to the directory holding setup.py because some python setups assume that you are
+    # in the same directory.
+    setup_dir = File.dirname(setup_data)
+
+    output = ::Dir.chdir(setup_dir) do
+      tmp = build_path("metadata.json")
+
+      toml_metadata_code = [
+        "from pyfpm_wheel import get_metadata_wheel",
+        "gmt = get_metadata_wheel.get_metadata_wheel('#{wheel_path}')",
+        "gmt.run('#{tmp}')"
+      ].join("\n")
+
+      get_metadata_cmd = "env PYTHONPATH=#{pylib}:$PYTHONPATH #{attributes[:python_bin]} " \
+        " -c " \
+        "#{Shellwords.escape(toml_metadata_code)}"
+
+      # @todo FIXME!
+      #      if attributes[:python_obey_requirements_txt?]
```

Comment:
Doesn't this break backward compatibility?

Reply:
It can.
Do you know any cmd option for PEP517 builds to do the same thing as --load-requirements-txt does for setup.py?